### ISSUE TITLE
Slow zoom & pan performance for line charts with over ~1k points onscreen

### ISSUE THREAD

**Issue by jrames** [danielgindi/Charts#2300]
I'm using a LineChartView with a single dataset containing about 4k points.  Zoomed in, it is performant.  When I zoom out substantially, it grinds to a halt.  In profiler, it appears that most time is being spent on context.strokePath() in drawLinear(context:dataset) of LineChartRenderer.swift.

I attempted the same thing in the Demo project using LineChart1ViewController.  I changed the X slider to be 4k and noticed zooming and panning effectively ground to a halt when set above 1k points when zoomed all the way out.   Zoomed in, it was of course fine.  This was with the default 'filled' option being turned off in the popup.  

I'm trying to figure out if there are any options I can toggle to improve performance.  I don't have much turned on right now - certainly no shading or gradients.  What else is especially time consuming to draw?  

The Android Wiki had mentioned this should be able to support charts up to about 10k points, so I can't help but wonder if what I'm doing is possible with the right configuration.

**Comment by liuxuan30** [comment]
I agree there is performance issues. It will redraw everything including calculation, which can be improved. 

You can check `draw()` to see what can be avoided if not necessary during scaling, if none of your data change. If `strokePath` is indeed the bottle neck, were you able to reduce the size? e.g. 4000 to 400 by sampling?
